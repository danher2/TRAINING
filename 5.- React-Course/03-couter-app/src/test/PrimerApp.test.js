import { render } from "@testing-library/react";
import '@testing-library/jest-dom';
import React from "react";
import PrimeraApp from "../PrimerApp";
import { shallow } from 'enzyme';

describe('Pruebas en componente <PrimerApp />', ()=>{

    // test('debe retornar el mensaje Hola soy Goku', ()=>{

    //     //given
    //     const saludo = 'Hola soy goku';
    //     const {getByText}= render(<PrimeraApp  saludo = {saludo}/>) // funcion que recibira un componente que queremos renderizar
        
    //     //when 


    //     //then
    //     expect(getByText(saludo)).toBeInTheDocument();


    // })

    test('debe de mostrar el compomente <PrimerApp /> correctamente',()=>{

        //given
        const saludo = 'Hola soy Goku';
        const wrapper = shallow(<PrimeraApp saludo= {saludo} />);

        //esperariamos que este componente se renderize de la forma correcta
        expect(wrapper).toMatchSnapshot();
    })




    test('debe de mostrar el subtitulo mostrado por props',()=>{

        //given
        const saludo = 'Hola soy Goku';
        const subtitulo = 'Soy un subtitulo';

        const wrapper = shallow(<PrimeraApp 
            saludo= {saludo} 
            subtitulo = {subtitulo}
            />
            );

            const textoParrafo = wrapper.find('p').text(subtitulo);
            console.log(textoParrafo);
        //esperariamos que este componente se renderize de la forma correcta
      expect(textoParrafo).toBe(subtitulo);


    })

})