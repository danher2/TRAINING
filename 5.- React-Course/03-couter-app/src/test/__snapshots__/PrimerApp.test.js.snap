// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en componente <PrimerApp /> debe de mostrar el compomente <PrimerApp /> correctamente 1`] = `
<Fragment>
  <h1>
    Hola soy Goku
    !!!
  </h1>
  <p>
    Soy un subtitulo
  </p>
</Fragment>
`;
